pkg/controllers/status/controller.go

```diff
@@ -422,19 +429,19 @@ func (s *KubeFedStatusController) reconcile(qualifiedName common.QualifiedName)
 			if apierrors.IsConflict(err) {
 				return worker.StatusConflict
 			}
-			runtime.HandleError(errors.Wrapf(err, "Failed to update status object for federated type %s %q", statusKind, key))
+			keyedLogger.WithValues("status-kind", statusKind).Error(err, "Failed to update status object")
 			return worker.StatusError
 		}
 	}
 
 	return worker.StatusAllOK
 }
 
-func (s *KubeFedStatusController) rawObjFromCache(store cache.Store, kind, key string) (pkgruntime.Object, error) {
+func (s *KubeFedStatusController) rawObjFromCache(ctx context.Context, store cache.Store, kind, key string) (pkgruntime.Object, error) {
 	cachedObj, exist, err := store.GetByKey(key)
+	keyedLogger := klog.FromContext(ctx)
 	if err != nil {
-		wrappedErr := errors.Wrapf(err, "Failed to query %s store for %q", kind, key)
-		runtime.HandleError(wrappedErr)
+		keyedLogger.WithValues("kind", kind).Error(err, "Failed to query object from store")
```

Comment:
It's preferred to propagate this error to the caller instead of logging it here.

Reply:
done, pls take review.